specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,429 @@
+        webView.CoreWebView2.SaveAsRequested -= WebView_SaveAsRequested;
+    else 
+        webView.CoreWebView2.SaveAsRequested += WebView_SaveAsRequested;
+    hasSaveAsRequestedEventHandler = !hasSaveAsRequestedEventHandler;
+    MessageBox.Show(hasSaveAsRequestedEventHandler? "Event Handler Added":"Event Handler Rremoved" , "Info");
+}
+
+void WebView_SaveAsRequested(object sender, CoreWebView2SaveAsRequestedEventArgs args)
+    {
+    // Hide the system default save as dialog.
+    args.SuppressDefaultDialog = true;
+
+    // Developer can obtain a deferral for the event so that the CoreWebView2
+    // doesn't examine the properties we set on the event args until
+    // after the deferral completes asynchronously.
+    CoreWebView2Deferral deferral = args.GetDeferral();
+
+    // We avoid potential reentrancy from running a message loop in the event
+    // handler. Show the customized dialog later when complete the deferral
+    // asynchronously.
+    System.Threading.SynchronizationContext.Current.Post((_) =>
+    {
+        using (deferral)
+        {
+            // This is a customized dialog example, the constructor returns after the 
+            // dialog interaction is completed by the end user.
+            var dialog = new SaveAsDialog(_Kinds);
+            if (dialog.ShowDialog() == true)
+            {
+                // Preview the content mime type, optional.
+                args.ContentMimeType;
+                
+                // Setting parameters of event args from this dialog is optional.
+                // The event args has default values.
+                args.ResultFilePath = dialog.Directory.Text + "/" + dialog.Filename.Text;
+                args.Kind = (CoreWebView2SaveAsRequestedKind)dialog.Kind.SelectedItem;
+                args.AllowReplace = (bool)dialog.AllowReplaceOldFile.IsChecked;
+            }
+            else
+            {
+                // Save As cancelled from this customized dialog
+                args.Cancel = true;
+            }
+        }
+    }, null);
+}
+```
+### Programmatic Save As
+Call RequestSaveAsAsync method to trigger the programmatic save as.
+```c#
+async void ProgrammaticSaveAsExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    CoreWebView2SaveAsRequestedResults result = await webView.CoreWebView2.RequestSaveAsAsync();
+    // Show RequestSaveAsAsync returned result, optional
+    MessageBox.Show(result.ToString(), "Info");
+}
+```
+
+# API Details
+## Win32 C++
+```c++
+/// Specifies save as requested kind selection options for `ICoreWebView2_20`,
+/// used in `SaveAsRequestedEventArgs`
+///
+/// When the source is a html page, supports to select `HTML_ONLY`,
+/// `SINGLE_FILE`, `COMPLETE`; when the source is a non-html,
+/// only allows to select `DEFAULT`; otherwise, will deny the download
+/// and return `COREWEBVIEW2_SAVE_AS_REQUESTED_KIND_NOT_SUPPORTED`.
+///
+/// The content type/format is a MIME type, indicated by the source
```

Comment:
I don't understand this paragraph. Do these refer to specific properties on the EventArgs? If so, please say the property name instead of 'content type/format' or "file's type or extension".